liolib.c

```diff
@@ -142,7 +142,7 @@ static int l_checkmode (const char *mode) {
 typedef luaL_Stream LStream;
 
 
-#define tolstream(L)	((LStream *)luaL_checkudata(L, 1, LUA_FILEHANDLE))
+#define tolstream(L)	(luaL_checktype(L, 1, LUA_TUSERDATA), (LStream *)lua_touserdata(L, 1))
```

Comment:
This introduces a security issue: now file methods can be passed an incorrect userdata object and incur C undefined behaviour.

Reply:
I will look into reverting this line because I don't think it ended up being used in practice anyway. We're only really using the `io.popen` function from this file right now.

I tried using more of this module to implement Lua 5.3-like `file:write()` behavior in `compat53.module` mode, but file handles returned by compat-enabled modules would be incompatible with `io` functions from non-compat-enabled caller code.

In the end, I managed to solve my problem without having to jump this hoop because apparently LuaJIT doesn't such a check, so I can make my modified metatable for files in `compat53.module` there. As of now, `file:write()` returns the file handle using `compat53.module` in all VMs except PUC-Rio Lua 5.1.